forecasting/src/autogluon/forecasting/dataset/ts_dataframe.py

```diff
@@ -0,0 +1,328 @@
+                ],
+                freq='D'
+            )
+
+        Returns:
+        --------
+        ts_df : TimeSeriesDataFrame
+            A data frame in TimeSeriesDataFrame format.
+        """
+
+        list_dataset.validate()
+        all_ts = []
+        for i, ts in enumerate(list_dataset):
+            start_timestamp = ts["start"]
+            target = ts["target"]
+            datetime_index = tuple(
+                pd.date_range(
+                    start_timestamp, periods=len(target), freq=list_dataset.freq
+                )
+            )
+            idx = pd.MultiIndex.from_product(
+                [(i,), datetime_index], names=["item_id", "timestamp"]
+            )
+            ts_df = pd.Series(target, name="target", index=idx).to_frame()
+            all_ts.append(ts_df)
+        return TimeSeriesDataFrame(pd.concat(all_ts))
+
+    @classmethod
+    def from_data_frame(cls, df: pd.DataFrame) -> TimeSeriesDataFrame:
+        """Convert a normal pd.DataFrame to a TimeSeriesDataFrame
+
+        Parameters:
+        -----------
+        df: pd.DataFrame
+            A pd.DataFrame with 'item_id' and 'timestamp' as columns. For example:
+
+               item_id  timestamp  target
+            0        0 2019-01-01       0
+            1        0 2019-01-02       1
+            2        0 2019-01-03       2
+            3        1 2019-01-01       3
+            4        1 2019-01-02       4
+            5        1 2019-01-03       5
+            6        2 2019-01-01       6
+            7        2 2019-01-02       7
+            8        2 2019-01-03       8
+
+        Returns:
+        --------
+        ts_df : TimeSeriesDataFrame
+            A data frame in TimeSeriesDataFrame format.
+        """
+        cls._validate_data_frame(df)
+        return TimeSeriesDataFrame(df.set_index(["item_id", "timestamp"]))
+
+    def split_by_time(
+        self, cutoff_time: pd.Timestamp
+    ) -> Tuple[TimeSeriesDataFrame, TimeSeriesDataFrame]:
+        """Split dataframe by a cutoff_time.
+
+        Parameters
+        ----------
+        cutoff_time : pd.Timestamp
+            The time to Split the current data frame into two data frames, all in TimeSeriesDataFrame format.
+
+        Returns
+        -------
+        data_before : TimeSeriesDataFrame
+            The first one after split contains time-series before the cutoff_time (exclude cutoff_time).
+        data_after : TimeSeriesDataFrame
+            The second one after split contains time-series after the cutoff_time (include cutoff_time).
+        """
+
+        nanosecond_before_cutoff = cutoff_time - pd.Timedelta(nanoseconds=1)
+        data_before = self.loc[(slice(None), slice(None, nanosecond_before_cutoff)), :]
+        data_after = self.loc[(slice(None), slice(cutoff_time, None)), :]
+        return TimeSeriesDataFrame(data_before), TimeSeriesDataFrame(data_after)
+
+    def split_by_item(
```

Comment:
Does it make sense to cutoff based off an int? Is this restrictive? Why not have `cutoff_items: list`? (I could be naive here)

Reply:
I see here needs more discussion actually. @RSNirwan also mentioned we may want to support item_id in categorical type or string. Also, we type of split options we will support for those types. Does it make sense that we take this as a follow up after the current PR has been merged?

Reply:
Why was this marked as resolved? Doesn't sound resolved from the comment. Happy to have a follow-up, but would prefer some kind of todo

Reply:
I saw you approved the change and considered that as an agreement for taking it as a follow up. For now I put a comment in the design doc so that we don't forget about it. We can also explicitly add a todo in the code if you wish.

Reply:
Ok no worries, so long as it is tracked in design doc its fine